### PR TITLE
feat(overlay): Allow to directly add sentry envelopes via trigger

### PR DESCRIPTION
**Pull request by mydea**
I am currently working on a hackweek project where I want to integrate the spotlight overlay into a chrome browser extension.

There, I cannot use sidecar (because I cannot launch a node service), but I can intercept envelopes sent to sentry myself. I'd like to be able to just run the spotlight overlay in the extension and directly push envelopes to it.

For this, I added a new hook, so you can add envelopes like this:

```
trigger("sentry:addEnvelope", sentryEnvelope);
```

### CODE REVIEW

**Review by Lms24 — APPROVED** [review]

**Comment by BYK** [comment]
Nice, this looks something like #133 I guess.

File: packages/overlay/src/integrations/sentry/index.ts [hunk 1]
```diff
@@ -49,8 +49,19 @@ export default function sentryIntegration(options: SentryIntegrationOptions = {}
 
       on('sentry:showError', onRenderError as EventListener);
 
+      const onAddEnvelope = (e: CustomEvent) => {
+        if (!e.detail.envelope) return;
+        processEnvelope({
+          contentType: 'application/x-sentry-envelope',
```
Comment (BYK) [inline packages/overlay/src/integrations/sentry/index.ts:55]:
```suggestion
          contentType: HEADER,
```

File: packages/overlay/src/integrations/sentry/index.ts [hunk 2]
```diff
@@ -49,8 +49,19 @@ export default function sentryIntegration(options: SentryIntegrationOptions = {}
 
       on('sentry:showError', onRenderError as EventListener);
 
+      const onAddEnvelope = (e: CustomEvent) => {
+        if (!e.detail.envelope) return;
+        processEnvelope({
+          contentType: HEADER,
+          data: e.detail.envelope,
```
Comment (BYK) [inline packages/overlay/src/integrations/sentry/index.ts:56]: Curious, why not just `e.detail` and pass the envelope as the event payload?

Comment (mydea) [reply packages/overlay/src/integrations/sentry/index.ts:56]: I basically just copied this from the `showError` scenario - but actually you are right, I think this is incorrect, will adjust it!

File: packages/overlay/src/integrations/sentry/index.ts [hunk 3]
```diff
@@ -41,16 +41,28 @@ export default function sentryIntegration(options: SentryIntegrationOptions = {}
         });
       }
 
-      const onRenderError = (e: CustomEvent) => {
-        log('Sentry Event', e.detail.event_id);
-        if (!e.detail.event) return;
-        sentryDataCache.pushEvent(e.detail.event).then(() => open(`/errors/${e.detail.event.event_id}`));
+      const onRenderError = (e: CustomEvent<{ event?: SentryEvent; event_id: string }>) => {
```
Comment (mydea) [inline packages/overlay/src/integrations/sentry/index.ts:44]: While at it I also typed this, for clarity - I think this is correct 😅  cc @BYK

File: packages/website/src/content/docs/reference/configuration.md [hunk 4]
```diff
@@ -155,7 +155,7 @@ import { trigger } from '@spotlightjs/spotlight';
 
 trigger('sentry:showError', {
   event: string,
-  eventId: string,
+  event_id: string,
```
Comment (mydea) [inline packages/website/src/content/docs/reference/configuration.md:158]: I think these docs were incorrect? We use `event_id` in the code, as far as I can see 😅

Comment (BYK) [reply packages/website/src/content/docs/reference/configuration.md:158]: Ugh, sorry :(

**Review by BYK — CHANGES_REQUESTED** [review]
Okay, so for this to work properly, we need to do this at a higher level, not inside Sentry integration: https://github.com/getsentry/spotlight/blob/86055ccd24f9980cace6e2286bea131bbccc50de/packages/overlay/src/sidecar.ts#L17

The name `sidecar.ts` is not great but essentially, that `listener` should be triggered which runs all integrations listening on the data and then updates their data with "setIntegrationData()`. This would allow bypassing sidecare much easier for other integrations too as it would provide a generic event ingestion system without the sidecar.

**Comment by BYK** [comment]
Pushed out #506 which should make doing the right thing in this patch much easier. LMK if you want to do it yourself or okay me doing it :)

**Comment by mydea** [comment]
If you can do it, I'd greatly appreciate it - I am rather busy right now so not sure when I can get around to it 😅

**Comment by BYK** [comment]
Should be done in #508. It still expects a full envelope in as a `string` or `Buffer`. Hoping that's a good-enough API? (directly analogous to the sidecar SSE format)

**Comment by mydea** [comment]
> Should be done in #508. It still expects a full envelope in as a `string` or `Buffer`. Hoping that's a good-enough API? (directly analogous to the sidecar SSE format)

That should work just fine I believe, thank you! :)

**Pull request closed without merging**